lib/puma/control_cli.rb

```diff
@@ -121,7 +121,7 @@ def prepare_configuration
         @control_url = sf.control_url
         @control_auth_token = sf.control_auth_token
         @pid = sf.pid
-      elsif @pidfile
+      elsif @pidfile && File.readable?(@pidfile)
```

Comment:
Why was this added?

Reply:
o/ this was added to avoid `No such file or directory @ rb_sysopen - tmp/pids/server.pid` when defined in `config/puma.rb` and have instead `Neither pid nor control url available` from https://github.com/puma/puma/blob/v3.12.1/lib/puma/control_cli.rb#L183

Reply:
I think this could be a separate PR-plus-tests, then.

Reply:
> I think this could be a separate PR-plus-tests, then.

First PR done in https://github.com/puma/puma/pull/1824